hera_cal/reflections.py

```diff
@@ -847,10 +861,22 @@ def interp_u(self, umodes, times, full_times=None, uflags=None, keys=None, overw
             # setup GP
             GP = gp.GaussianProcessRegressor(kernel=kernel, optimizer=optimizer, normalize_y=False)
 
+            # setup regression data
+            X = times - Xmean
+            Y = umodes[k][:, :Ninterp].copy()
+            F = uflags[k].copy()
+
+            # mirror if desired
+            if Nmirror > 0:
+                assert Nmirror < X.size, "Nmirror can't be equal or larger than Ntimes"
+                _, X = vis_clean.zeropad_array(Y, binvals=X, zeropad=Nmirror, axis=0)
+                Y = np.concatenate([Y[1:Nmirror + 1, :][::-1, :], Y, Y[-Nmirror - 1:-1, :][::-1, :]], axis=0)
```

Comment:
I think you can do this much more cleanly with `np.pad` with the `'reflect'` option.

Reply:
I think `np.pad` doesn't have an `axis` kwarg, so it can only pad all axes of an array, whereas I need to pad only a single axis here

Reply:
You can, actually. The `pad_width` kwarg takes a list of tuples: ((before_1, after_1), … (before_N, after_N))

So just set the axis padding width on the axes you don't want to pad to 0.

Reply:
ah interesting. except that I'm discovering that it doesn't do this well for boolean arrays, so I'll have to do what I have here anyways for the flag array, so might as well be consistent between data and flags..